util/oc_features.h

```diff
@@ -67,4 +67,12 @@
 #define OC_HAS_FEATURE_ETAG_INTERFACE
 #endif /* OC_HAS_FEATURE_ETAG && OC_STORAGE */
 
+#if defined(OC_DYNAMIC_ALLOCATION) && !defined(OC_INOUT_BUFFER_SIZE)
+#define OC_HAS_FEATURE_MESSAGE_SHRINK_BUFFER
```

Comment:
I'm thinking `OC_HAS_FEATURE_MESSAGE_DYNAMIC_BUFFER` would be better if we use it to ifdef not only the shrink function.

Reply:
Renamed